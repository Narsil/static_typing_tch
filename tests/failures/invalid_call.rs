use static_typing_tch::tensor_check;
use tch::{kind::Kind, Device, Tensor};

tensor_check! {
fn gelu(x: &Tensor<(B, S, H)>) -> Tensor<(B, S, H)> {
    let y: Tensor = 0.79788456 * x * (1.0 + 0.044715 * x * x);
    x * 0.5 * (1.0 + y.tanh())
}


fn main(){
    let hidden_states = Tensor::ones(&[3, 24], (Kind::Float, Device::Cpu));
    let _hidden_states = gelu(&hidden_states);
}
}
