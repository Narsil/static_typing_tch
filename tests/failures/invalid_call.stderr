error: This call is invalid, expected Declared(
           TensorType {
               shape: [
                   Symbol(
                       "B",
                   ),
                   Symbol(
                       "S",
                   ),
                   Symbol(
                       "H",
                   ),
               ],
               kind: Implicit,
               device: Implicit,
           },
       ) but got Inferred(
           TensorType {
               shape: [
                   Value(
                       3,
                   ),
                   Value(
                       24,
                   ),
               ],
               kind: Float,
               device: Cpu,
           },
       )
  --> tests/failures/invalid_call.rs:13:31
   |
13 |     let _hidden_states = gelu(&hidden_states);
   |                               ^^^^^^^^^^^^^^
